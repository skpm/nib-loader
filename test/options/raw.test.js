/* eslint-disable
  prefer-destructuring,
*/
import path from 'path';
import webpack from '../helpers/compiler';

describe('Options', () => {
  describe('raw', () => {
    test('{String}', async () => {
      const config = {
        loader: {
          test: /(png|jpg|svg)/,
          options: {
            raw: true,
            outputPath: value => path.join('..', 'Resources', '_webpack_resources', value),
            publicPath: value => `"file://" + context.plugin.urlForResourceNamed("_webpack_resources/${value}").path()`,
          },
        },
      };

      const stats = await webpack('fixture.js', config);
      const { assets, source } = stats.toJson().modules[1];

      expect({ assets, source }).toMatchSnapshot();
    });
  });
});
