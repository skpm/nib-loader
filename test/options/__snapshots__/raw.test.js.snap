// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options raw {String} 1`] = `
Object {
  "assets": Array [
    "../Resources/_webpack_resources/9c87cbf3ba33126ffd25ae7f2f6bbafb.png",
  ],
  "source": "module.exports = \\"file://\\" + context.plugin.urlForResourceNamed(\\"_webpack_resources/9c87cbf3ba33126ffd25ae7f2f6bbafb.png\\").path();",
}
`;
